contracts/crowdsale/distribution/PostDeliveryCrowdsale.sol

```diff
@@ -32,12 +40,24 @@ contract PostDeliveryCrowdsale is TimedCrowdsale {
     }
 
     /**
-     * @dev Overrides parent by storing balances instead of issuing tokens right away.
+     * @dev Overrides parent by storing due balances, and delivering tokens to the vault instead of the end user. This
+     * ensures that the tokens will be available by the time they are withdraw (which may not be the case if
+     * _deliverTokens was called later).
      * @param beneficiary Token purchaser
      * @param tokenAmount Amount of tokens purchased
      */
     function _processPurchase(address beneficiary, uint256 tokenAmount) internal {
         _balances[beneficiary] = _balances[beneficiary].add(tokenAmount);
+        _deliverTokens(address(_vault), tokenAmount);
     }
+}
 
+/**
+ * @title TokenVault
+ * @dev Similar to an Escrow for tokens, this contract allows its primary account to spend its tokens as it sees fit.
+ */
+contract TokenVault is Secondary {
```

Comment:
Thoughts about naming this `__unstable__TokenVault`? :slightly_smiling_face:

Reply:
Ideally we wouldn't even export it, but some tools (e.g. 0x-coverage) require each contract to be its own separate file. Would the file also be called that in such a scenario?

Reply:
We don't use 0x-coverage.